### PR TITLE
refactor(typescript): added typescript endpoint declarations

### PR DESCRIPTION
**Pull request by kleyow**

### CODE REVIEW

File: src/lib/requests/baseRequests.js [hunk 1]
```diff
@@ -88,7 +89,10 @@ class BaseRequests {
             thirdparty: formatEndpointOrDefault(config.thirdpartyRequestsEndpoint, this.transportScheme, this.peerEndpoint),
         };
 
-        this.wso2Auth = config.wso2Auth;
+        this.wso2Auth = new WSO2Auth({
```
Comment (kleyow) [inline src/lib/requests/baseRequests.js:92]: Believe this should be a `WSO2Auth` object and not what is passed through config.
I believe logger and config.wso2Auth are the only things that should be passed through. lmk.

Comment (kleyow) [reply src/lib/requests/baseRequests.js:92]: Hmm I see that it should be passed a `tlsCreds` object. but which one do I pass? `inbound`, `outbound` or `test`

`inbound`, `outbound` or `test` being referenced from https://github.com/mojaloop/sdk-scheme-adapter/blob/master/src/config.js#L34

Comment (lewisdaly) [reply src/lib/requests/baseRequests.js:92]: Hmm this looks scary to me - perhaps we need to add tests for the `BaseRequests` constructor to prove that either it hasn't changed with this addition, or that there was a bug that you fixed

Comment (kleyow) [reply src/lib/requests/baseRequests.js:92]: I agree. I'll add tests after some more input.

Seems to have been changed in this commit.
https://github.com/mojaloop/sdk-standard-components/commit/7809dabca72c18af5b88cddd3a8e4a1993bc98e4#diff-646394f9756db656295cb82e958875c4L73

I do know that this line uses `getToken` which is a function on `WSO2Auth`
https://github.com/mojaloop/sdk-standard-components/blob/5f9c3e61a99069a0ae87b135f9fd92aa9301af7b/src/lib/requests/baseRequests.js#L290

Comment (kleyow) [reply src/lib/requests/baseRequests.js]: @lewisdaly I've unstaged the wso2 changes for now. Let me check in the typescript then we can circle back to this wso2 issue(?)

**Review by eoln — APPROVED** [review]
lgtm

**Review by eoln — APPROVED** [review]

**Review by lewisdaly — APPROVED** [review]

**Pull request merged**